src/main.rs

```diff
@@ -56,72 +48,107 @@ fn main() {
-        .ok_or(Error::new(ErrorKind::InvalidInput, "Cannot get a batch reference from the file name"))?;
-    let output_batch_reference = &input_batch_reference.replace("TDR", "TST");
-
-    let extracted_output_original_name = dir_output.join(PathBuf::from(&input_batch_reference));
-    let extracted_output_path = dir_output.join(PathBuf::from(output_batch_reference));
+    let output_tar_gz_path: PathBuf =
+        Path::new(&dir_output).join(Path::new(&tar_gz_file_name.replace("TDR", "TST")));
+    let uncompressed_folder_input_path: &PathBuf = &file.with_extension("").with_extension("");
+    let input_batch_reference: &str = uncompressed_folder_input_path
+        .file_name()
+        .and_then(|name| name.to_str())
+        .ok_or(Error::new(
+            ErrorKind::InvalidInput,
+            "Cannot get a batch reference from the file name",
+        ))?;
+    let output_batch_reference: &String = &input_batch_reference.replace("TDR", "TST");
+
+    let extracted_output_original_name: PathBuf =
+        dir_output.join(PathBuf::from(&input_batch_reference));
+    let extracted_output_path: PathBuf = dir_output.join(PathBuf::from(output_batch_reference));
 
     fs::create_dir_all(extracted_output_path.clone())?;
 
     decompress_file(&file, &dir_output)?;
 
-    let metadata_input_file_path = &extracted_output_path.join(PathBuf::from(format!("TRE-{}-metadata.json", &input_batch_reference)));
-    let metadata_output_file_path = &extracted_output_path.join(PathBuf::from(format!("TRE-{}-metadata.json", &output_batch_reference)));
+    let metadata_input_file_path: &PathBuf = &extracted_output_path.join(PathBuf::from(format!(
+        "TRE-{input_batch_reference}-metadata.json"
+    )));
+    let metadata_output_file_path: &PathBuf = &extracted_output_path.join(PathBuf::from(format!(
+        "TRE-{output_batch_reference}-metadata.json"
+    )));
 
     if extracted_output_path.exists() {
         fs::remove_dir_all(&extracted_output_path)?;
     }
     fs::rename(&extracted_output_original_name, &extracted_output_path)?;
     fs::rename(&metadata_input_file_path, &metadata_output_file_path)?;
 
-    let mut json_value = parse_json(&metadata_output_file_path)?;
+    let mut metadata_json_value: Value = parse_metadata_json(&metadata_output_file_path)?;
 
-    let docx_file_name = json_value["parameters"]["TRE"]["payload"]["filename"].as_str()
+    let docx_file_name: &str = metadata_json_value["parameters"]["TRE"]["payload"]["filename"]
+        .as_str()
         .ok_or("Filename is missing from the metadata json")
         .map_err(|e| Error::new(ErrorKind::InvalidInput, e))?;
 
-    let judgment_name = json_value["parameters"]["PARSER"]["name"].to_string();
-    let docx_path_string = extracted_output_path.join(PathBuf::from(docx_file_name));
+    let judgment_name: String = metadata_json_value["parameters"]["PARSER"]["name"].to_string();
+    let docx_path_string: PathBuf = extracted_output_path.join(PathBuf::from(docx_file_name));
```

Comment:
Should this be just `docx_path`?

Reply:
It should, good spot. It used to be a string before I realised using PathBuf was better.